lib/Controller/LoginController.php

```diff
@@ -356,8 +357,20 @@ public function code($state = '', $code = '', $scope = '') {
 		$this->logger->debug('Logging user in');
 
 		$this->userSession->setUser($user);
-		$this->userSession->completeLogin($user, ['loginName' => $user->getUID(), 'password' => '']);
+		
+		$this->userSession->getSession()->regenerateId();
+		$tokenProvider = \OC::$server->query(DefaultTokenProvider::class);
+		$this->userSession->setTokenProvider($tokenProvider);
+		
 		$this->userSession->createSessionToken($this->request, $user->getUID(), $user->getUID());
+		
+		$token = $tokenProvider->getToken($this->userSession->getSession()->getId());
+
+		$this->userSession->completeLogin($user, [
+			'loginName' => $user->getUID(),
+			'password' => '',
```

Comment:
Might be that we should rather fix this behavior in the server or differently then as the empty password approach is also used for SAML and hasn't shown any issue there as well.